code/Helper/Data.php

```diff
@@ -511,6 +512,65 @@ public function rebuildStoreProductIndexPage($storeId, $collectionDefault, $page
         $this->logger->stop('rebuildStoreProductIndexPage '.$this->logger->getStoreName($storeId).' page ' . $page . ' pageSize '. $pageSize);
     }
 
+    /**
+     * Removes products that are no longer indexable, i.e. they are disabled,
+     * from the specified Algolia index. You can provide an array from product
+     * IDs that you want to query. Otherwise, the entire product collection will
+     * be queried.
+     *
+     * @param string $storeId
+     * @param array $productIds
+     */
+    public function removeNonIndexableProducts($storeId, array $productIds = array())
+    {
+        $this->logger->start('REMOVE FROM ALGOLIA');
+
+        if (!$this->config->getShowOutOfStock()) {
```

Comment:
Yes I would rather add this condition to the collection creation.

A bit like this => https://github.com/algolia/algoliasearch-magento/blob/master/code/Helper/Entity/Producthelper.php#L82-L83